File: Day 23 - Binary Tree/Maximum depth of binary tree.cpp
class Solution
{
public:
  int maxDepth(Node *root)
  {
    if (root == NULL)
      return 0;
    int lh = maxDepth(root->left);
    int rh = maxDepth(root->right);
    return max(lh, rh) + 1;
  }
};

File: Day 23 - Binary Tree/Maximum depth of binary tree.java
class Solution {
  public static int maxDepth(Node root) {

        if(root == null) return 0;
        
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        
        return Math.max(left, right) + 1;
  }
}
 